tests/contrib/langchain/test_langchain_community.py

```diff
@@ -1276,6 +1278,169 @@ def test_faiss_vectorstore_retrieval(langchain_community, langchain_openai, requ
+    )
+    llm = langchain_openai.ChatOpenAI(async_client=client)
+    parser = langchain_core.output_parsers.StrOutputParser()
+
+    chain = prompt | llm | parser
+    async for _ in chain.astream({"input": "how can langsmith help with testing?"}):
+        pass
+
+
+@pytest.mark.snapshot(
+    ignores=IGNORE_FIELDS,
+    token="tests.contrib.langchain.test_langchain_community.test_streamed_chat",
+)
+async def test_astreamed_chat(langchain_openai, async_streamed_response_responder):
+    client = async_streamed_response_responder(
+        module="openai",
+        client_class_key="AsyncOpenAI",
+        http_client_key="http_client",
+        endpoint_path=["chat", "completions"],
+        file="lcel_openai_chat_streamed_response.txt",
+    )
+
+    model = langchain_openai.ChatOpenAI(async_client=client)
+
+    async for _ in model.astream(input="how can langsmith help with testing?"):
+        pass
+
+
+@pytest.mark.snapshot(
+    ignores=IGNORE_FIELDS,
+    token="tests.contrib.langchain.test_langchain_community.test_streamed_llm",
+)
+async def test_astreamed_llm(langchain_openai, async_streamed_response_responder):
+    client = async_streamed_response_responder(
+        module="openai",
+        client_class_key="AsyncOpenAI",
+        http_client_key="http_client",
+        endpoint_path=["completions"],
+        file="lcel_openai_llm_streamed_response.txt",
+    )
+
+    llm = langchain_openai.OpenAI(async_client=client)
+
+    async for _ in llm.astream(input="How do I write technical documentation?"):
+        pass
+
+
+@pytest.mark.snapshot(ignores=IGNORE_FIELDS)
+def test_streamed_json_output_parser(langchain, langchain_core, langchain_openai, streamed_response_responder):
+    client = streamed_response_responder(
+        module="openai",
+        client_class_key="OpenAI",
+        http_client_key="http_client",
+        endpoint_path=["chat", "completions"],
+        file="lcel_openai_chat_streamed_response_json_output_parser.txt",
+    )
+
+    model = langchain_openai.ChatOpenAI(model="gpt-4o", max_tokens=50, client=client)
+    parser = langchain_core.output_parsers.JsonOutputParser()
+
+    chain = model | parser
+    inp = (
+        "output a list of the country france their population in JSON format. "
+        'Use a dict with an outer key of "countries" which contains a list of countries. '
+        "Each country should have the key `name` and `population`"
+    )
+
+    messages = [
+        langchain.schema.SystemMessage(content="You know everything about the world."),
+        langchain.schema.HumanMessage(content=inp),
+    ]
+
+    for _ in chain.stream(input=messages):
+        pass
+
+
+# until we fully support `astream_events`, we do not need a snapshot here
+# this is just a regression test to make sure we don't throw
+async def test_astreamed_events_does_not_throw(langchain_openai, langchain_core, async_streamed_response_responder):
```

Comment:
So to confirm, we do not trace astream_events() (there is also no `stream_events()`, just `astream_events()`?), but it does in fact interfere with `stream()` so possibly can be affected by our patching of `stream()`?

Could we add a note in the PR description and a JIRA item to follow up on this?

Reply:
yep that's my understanding of how it works, and yes no `stream_events`. I'll make a note + follow up ticket